analyze_foldamers/parameters/secondary_structure.py

```diff
@@ -0,0 +1,679 @@
+from openmmtools.multistate import MultiStateReporter, ReplicaExchangeAnalyzer
+import pymbar
+
+kB = unit.MOLAR_GAS_CONSTANT_R # Boltzmann constant
+
+def get_native_contacts(cgmodel, native_structure, native_contact_distance_cutoff):
+    """
+        Given a coarse grained model, positions for that model, and positions for the native structure, this function calculates the fraction of native contacts for the model.
+
+        :param cgmodel: CGModel() class object
+        :type cgmodel: class
+
+        :param native_structure: Positions for the particles in a coarse grained model.
+        :type native_structure: np.array( float * unit.angstrom ( num_particles x 3 ) )
+
+        :param native_contact_distance_cutoff: The maximum distance for two nonbonded particles that are defined as "native",default=None
+        :type native_contact_distance_cutoff: `Quantity() <https://docs.openmm.org/development/api-python/generated/simtk.unit.quantity.Quantity.html>`_
+
+        :returns:
+          - native_contact_list - A list of the nonbonded interactions whose inter-particle distances are less than the 'native_contact_cutoff_distance'.
+
+        """
+
+    nonbonded_interaction_list = cgmodel.nonbonded_interaction_list
+    native_structure_distances = distances(nonbonded_interaction_list, native_structure)
+    native_contact_list = []
+
+    for interaction in range(len(nonbonded_interaction_list)):
+        if native_structure_distances[interaction].__lt__(native_contact_distance_cutoff):
+            native_contact_list.append(nonbonded_interaction_list[interaction])
+
+    return native_contact_list
+
+
+def get_number_native_contacts(cgmodel, native_structure, native_contact_distance_cutoff):
+    """
+        Given a coarse grained model, positions for that model, and positions for the native structure, this function calculates the fraction of native contacts for the model.
+
+        :param cgmodel: CGModel() class object
+        :type cgmodel: class
+
+        :param native_structure: Positions for the particles in a coarse grained model.
+        :type native_structure: np.array( float * unit.angstrom ( num_particles x 3 ) )
+
+        :param native_contact_cutoff_distance: The maximum distance for two nonbonded particles that are defined as "native",default=None
+        :type native_contact_cutoff_distance: `Quantity() <https://docs.openmm.org/development/api-python/generated/simtk.unit.quantity.Quantity.html>`_
+
+        :returns:
+          - contacts - The number of nonbonded interactions that are considered 'native'.
+
+        """
+
+    native_contact_list = get_native_contacts(
+        cgmodel, native_structure, native_contact_distance_cutoff
+    )
+
+    contacts = len(native_contact_list)
+
+    return contacts
+
+def expectations_fraction_contacts(list_native_contacts, native_pdb, distance_cutoff, temperature_list, output_directory="output", output_data="output.nc", num_intermediate_states=0):
```

Comment:
This function should be moved into cg_openmm, since it involves calculating thermodynamic expectations, and requires a Boltzmann weighted ensemble as one would get from a simulation.